DCA/adapter.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+
+from PIL import Image
+import argparse
+from pathlib import Path
+
+from adapter_base import AdapterBase
+from utils.tools import *
+from utils.my_tools import *
+from module.Encoder import Deeplabv2
+from albumentations import Normalize
+from albumentations import *
+import ever as er
+
+def max_power_of_2(num : int):
+    p: int = 2
+    while p * 2 <= num:
+        p *= 2
+    return p
+
+
+class DcaAdapter(AdapterBase):
+    def __init__(self, size=1024, factor=2, mode="Urban"):
+        self.size = size
+        self.factor = factor
+        self.mode = mode
+
+    def set_size(self, size):
+        self.size = size
+
+    def process(self, image: np.ndarray):
+        # image - RGB
+        image = self._transform_image(image)
+        model = self._build_model()
+        raw_predictions = self._process(model, image)
+        predictions = self._postprocess_predictions(raw_predictions)
+        return predictions
+
+    # requires MxMx3 image
+    def _transform_image(self, image: np.ndarray):
+        image = Image.fromarray(image)
+        image = image.convert("RGB")
+        image = image.resize((self.size, self.size))
+        image = np.array(image)
+        print(image.shape)
+        mask = np.zeros([512, 512])
+        transformer = Compose([
+            Normalize(mean=(73.53223948, 80.01710095, 74.59297778),
```

Comment:
[Why magic numbers are bad](https://stackoverflow.com/questions/47882/what-are-magic-numbers-and-why-do-some-consider-them-bad)

Reply:
It was in author's code, so I have no idea where these numbers are from

Reply:
Not the best argument. You need to understand why they are needed and remove the magic numbers. How can this code be maintained if even the author does not know what these numbers mean?